tests/src/OneLogin/saml2_tests/response_test.py

```diff
@@ -549,11 +519,9 @@ def testValidateID(self):
         settings = OneLogin_Saml2_Settings(self.loadSettingsJSON())
         xml = self.file_contents(join(self.data_path, 'responses', 'invalids', 'no_id.xml.base64'))
         response = OneLogin_Saml2_Response(settings, xml)
-        try:
-            valid = response.is_valid(self.get_request_data())
-            self.assertFalse(valid)
-        except Exception as e:
-            self.assertEqual('Missing ID attribute on SAML Response', str(e))
```

Comment:
When I was refactoring these tests I didn't take into consideration the changes that would come from adding the new argument. I wanted to keep the cleaning changes separate from the new feature to keep the commits simple.

I don't have any strong opinion about whether we should run these tests with the raising version. The default code calls these functions without the raises being set, but with the raises set we can more precisely make sure that the function failed with the wanted error.

I can change these functions to use the exception raising if you prefer that.